src/s3/util/fileViewerManager.ts

```diff
@@ -78,86 +92,135 @@ export class S3FileViewerManager {
                     return
                 }
                 await this.removeAndCloseTab(activeTab)
-                await this.openTab(fileNode)
+                await this.openInReadMode(fileNode)
                 this._onDidChange.fire(activeTab.s3Uri)
             }
         })
+    }
 
-        this.window.onDidChangeVisibleTextEditors(editors => {
-            const editorSet = new Set(editors)
-            for (const value of this.activeTabs.values()) {
-                //if visible text editors don't contain a given S3Tab anymore,
-                //set the S3Tab.editor to undefined
-                if (value.editor) {
-                    if (!editorSet.has(value.editor)) {
-                        value.editor = undefined
-                    }
-                }
-            }
-        })
+    public async focusAndCloseTab(
+        uri: vscode.Uri,
+        editor?: vscode.TextEditor,
```

Comment:
Just so I'm clear, under what circumstances is `editor` not defined? Would it result in opening a file that isn't open, only to immediately close the tab?

Reply:
It is just to be sure it gets closed in case it was set undefined